flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -1066,26 +1069,23 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
 
 			coord.startCheckpointScheduler();
 
-			//trigger first checkpoint
-			Thread.sleep(100);
-
-			assertEquals(1, numCalls.get());
-
-			//no new checkpoint has been triggered
-			Thread.sleep(100);
-			assertEquals(1, numCalls.get());
-
-			//no new checkpoint has been triggered
-			Thread.sleep(100);
-			assertEquals(1, numCalls.get());
-
-			//new checkpoint has been triggered
-			Thread.sleep(300);
-			assertEquals(2, numCalls.get());
+			for (int x=0; x<20; x++) {
+				Thread.sleep(100);
+				if (numCalls.get() > 0) {
```

Comment:
Since you are waiting on this condition, I would add a check after the loop with a good error message if <= 0 in case that the call did not happen in 20 * 100 ms.